apollo-router/src/spec/query/change.rs

```diff
@@ -33,14 +33,12 @@ pub(crate) const JOIN_TYPE_DIRECTIVE_NAME: &str = "join__type";
 pub(crate) struct QueryHashVisitor<'a> {
```

Comment:
This file needs an extensive rust doc.

Describe the algorithm, and details of why the special literals are needed.

Reply:
done in 66c9a0001